javalanglib/src/main/scala/java/lang/Enum.scala

```diff
@@ -0,0 +1,34 @@
+package java.lang
+
+abstract class Enum[E <: Enum[E]] protected (_name: String, _ordinal: Int)
+    extends Comparable[E] with java.io.Serializable {
+
+  final def name(): String = _name
+
+  final def ordinal(): Int = _ordinal
+
+  override def toString(): String = _name
+
+  override final def equals(that: Any): scala.Boolean = that match {
+    case that: Enum[E] if that.getClass == getClass =>
+      _ordinal == that.ordinal() && _name == that.name()
+
+    case _ => false
+  }
+
+  override final def hashCode(): Int =
+    31 * _name.hashCode() + _ordinal.hashCode()
```

Comment:
Enum values are only ever equal to strictly themselves, so `equals` and `hashCode` can simply call `super.{equals,hashCode}` to inherit `Object`'s definitions (while still being `final`). I recommend marking them `@inline`:
```scala
@inline
override final def equals(that: Any): scala.Boolean = super.equals(that)

@inline
override final def hashCode(): Int = super.hashCode()
```

Although the JavaDoc spec is a bit scarce, experimentation in a REPL suggests that this is also what the JDK does.